samples/MvcSandbox/Startup.cs

```diff
@@ -14,7 +14,7 @@ public class Startup
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc();
+            services.AddMvc().SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_1);
```

Comment:
This is just a design PR, I didn't fill in the details or tests yet because I want to get feedback on the overall direction.

Reply:
Undo